dattri/benchmark/load.py

```diff
@@ -0,0 +1,268 @@
+                the difference is the dataset sampling (half sampling) for
+                each model checkpoint.
+            - "train_dataset": The path to the training dataset with the
+                same order as the groundtruth's indices.
+            - "test_dataset": The path to the testing dataset with the
+                same order as the groundtruth's indices.
+            - "loss_func": The loss function for the model training. Normally
+                speaking, this should be the same as the target function.
+            - "target_func": The target function for the data attribution. Normally
+                speaking, this should be the same as the loss function.
+            - "train_func": The training function for the model. Normally it's
+                not required if the pre-trained model checkpoints is enough
+                for the algorithm you want to benchmark.
+
+            The second tuple contains the ground truth for the benchmark,
+            the items are listed as following. It can be directly sent to the
+            metrics function defined in `dattri.metrics`. Notably, the groundtruth
+            depends on the `metric` parameter user stated.
+
+    Raises:
+        ValueError: If the model or dataset is not supported.
+    """
+    if model not in SUPPORTED_MODELS:
+        error_msg = f"Model {model} is not supported,\
+                      please choose from {SUPPORTED_MODELS}."
+        raise ValueError(error_msg)
+
+    if dataset not in SUPPORTED_DATASETS:
+        error_msg = f"Dataset {dataset} is not supported,\
+                      please choose from {SUPPORTED_DATASETS}."
+        raise ValueError(error_msg)
+
+    identifier = f"{dataset}_{model}"
+    url_map = generate_url_map(identifier)
+    download_path = pathlib.Path(download_path).expanduser()
+
+    if not (download_path / identifier).exists() or redownload:
+        for key in ["models_full", "models_half"]:
+            _download(
+                url_map[key],
+                download_path / "benchmark" / identifier / key,
+                unzip=True,
+            )
+        for path in url_map["groundtruth"]["lds"]:
+            _download(
+                path,
+                download_path / "benchmark" / identifier / "lds",
+                unzip=False,
+                file_name=path.split("?")[0].split("/")[-1],
+            )
+        for path in url_map["groundtruth"]["loo"]:
+            _download(
+                path,
+                download_path / "benchmark" / identifier / "loo",
+                unzip=False,
+                file_name=path.split("?")[0].split("/")[-1],
+            )
+
+    models_full_count = _count_folders(
+        download_path / "benchmark" / identifier / "models_full",
+    )
+    models_full_list = [
+        download_path
+        / "benchmark"
+        / identifier
+        / "models_full"
+        / f"{i}"
+        / f"model_weights_{i}.pt"
```

Comment:
For our retraining script, I remember the pt's name will be `model_weights_0.pt` if we set `num_runs_per_subset` to 1. So I guess it is possible to have multiple checkpoint files in `f"{i}/` with different names.

Please correct me if I am wrong.

Reply:
Good catch. I directly change them to `model_weights_0.pt`. These are the models trained by full/half datasets. So only 1 model will be put to each folder.